patches/clang/0001-OpenCL-3.0-feature-macro-support.patch

```diff
@@ -0,0 +1,3176 @@
+ void __ovld vstore16(half16 data, size_t offset, __private half *p);
+ #endif //cl_khr_fp16
+-#endif //defined(__OPENCL_CPP_VERSION__) || (__OPENCL_C_VERSION__ >= CL_VERSION_2_0)
++#endif //__opencl_c_generic_address_space
+ 
+ /**
+  * Read sizeof (half) bytes of data from address
+@@ -11881,7 +11881,7 @@ float16 __ovld vload_half16(size_t offset, const __private half *p);
+  * The default current rounding mode is round to
+  * nearest even.
+  */
+-#if defined(__OPENCL_CPP_VERSION__) || (__OPENCL_C_VERSION__ >= CL_VERSION_2_0)
++#ifdef __opencl_c_generic_address_space
+ void __ovld vstore_half(float data, size_t offset, half *p);
+ void __ovld vstore_half_rte(float data, size_t offset, half *p);
+ void __ovld vstore_half_rtz(float data, size_t offset, half *p);
+@@ -11927,7 +11927,7 @@ void __ovld vstore_half_rtz(double data, size_t offset, __private half *p);
+ void __ovld vstore_half_rtp(double data, size_t offset, __private half *p);
+ void __ovld vstore_half_rtn(double data, size_t offset, __private half *p);
+ #endif //cl_khr_fp64
+-#endif //defined(__OPENCL_CPP_VERSION__) || (__OPENCL_C_VERSION__ >= CL_VERSION_2_0)
++#endif //__opencl_c_generic_address_space
+ 
+ /**
+  * The floatn value given by data is converted to
+@@ -11940,7 +11940,7 @@ void __ovld vstore_half_rtn(double data, size_t offset, __private half *p);
+  * The default current rounding mode is round to
+  * nearest even.
+  */
+-#if defined(__OPENCL_CPP_VERSION__) || (__OPENCL_C_VERSION__ >= CL_VERSION_2_0)
++#ifdef __opencl_c_generic_address_space
+ void __ovld vstore_half2(float2 data, size_t offset, half *p);
+ void __ovld vstore_half3(float3 data, size_t offset, half *p);
+ void __ovld vstore_half4(float4 data, size_t offset, half *p);
+@@ -12146,7 +12146,7 @@ void __ovld vstore_half4_rtn(double4 data, size_t offset, __private half *p);
+ void __ovld vstore_half8_rtn(double8 data, size_t offset, __private half *p);
+ void __ovld vstore_half16_rtn(double16 data, size_t offset, __private half *p);
+ #endif //cl_khr_fp64
+-#endif //defined(__OPENCL_CPP_VERSION__) || (__OPENCL_C_VERSION__ >= CL_VERSION_2_0)
++#endif //__opencl_c_generic_address_space
+ 
+ /**
+  * For n = 1, 2, 4, 8 and 16 read sizeof (halfn)
+@@ -13482,9 +13482,6 @@ long __ovld atomic_fetch_min_explicit(volatile atomic_long *object, long operand
+ ulong __ovld atomic_fetch_min(volatile atomic_ulong *object, ulong operand);
+ ulong __ovld atomic_fetch_min_explicit(volatile atomic_ulong *object, ulong operand, memory_order order);
+ ulong __ovld atomic_fetch_min_explicit(volatile atomic_ulong *object, ulong operand, memory_order order, memory_scope scope);
+-ulong __ovld atomic_fetch_min(volatile atomic_ulong *object, long operand);
+-ulong __ovld atomic_fetch_min_explicit(volatile atomic_ulong *object, long operand, memory_order order);
+-ulong __ovld atomic_fetch_min_explicit(volatile atomic_ulong *object, long operand, memory_order order, memory_scope scope);
```

Comment:
Why you are removing these builtins?

Reply:
That's definitely  a problem after applying patch from other repo. Thanks for noticing.